main/SAPI.c

```diff
@@ -209,15 +209,22 @@ static void sapi_read_post_data(void)
 		SG(request_info).post_entry = NULL;
 		if (!sapi_module.default_post_reader) {
 			/* no default reader ? */
-			SG(request_info).content_type_dup = NULL;
+			if (SG(request_info).content_type_dup) {
+				efree(SG(request_info).content_type_dup);
+				SG(request_info).content_type_dup = NULL;
+			}
 			sapi_module.sapi_error(E_WARNING, "Unsupported content type:  '%s'", content_type);
+			efree(content_type);
```

Comment:
I think this is actually the only part that fixes a real leak.

Reply:
It might be a good idea to try to write a test for this and separate from the previous checks